bool access_granted
string user_name
